x-pack/plugins/ml/public/alerting/anomaly_detection_rule/advanced_settings.tsx

```diff
@@ -17,16 +14,28 @@ import {
   EuiSpacer,
   EuiText,
 } from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { isDefined } from '@kbn/ml-is-defined';
+import type { FC } from 'react';
+import React, { useMemo } from 'react';
+import { validateTopNBucket } from '../validators';
+import { TOP_N_BUCKETS_COUNT } from '../../../common/constants/alerts';
 import { type MlAnomalyDetectionAlertAdvancedSettings } from '../../../common/types/alerts';
 import { TimeIntervalControl } from '../time_interval_control';
-import { TOP_N_BUCKETS_COUNT } from '../../../common/constants/alerts';
 
 interface AdvancedSettingsProps {
   value: MlAnomalyDetectionAlertAdvancedSettings;
   onChange: (update: Partial<MlAnomalyDetectionAlertAdvancedSettings>) => void;
 }
 
 export const AdvancedSettings: FC<AdvancedSettingsProps> = React.memo(({ value, onChange }) => {
+  const topNBucketsValidationErrors = useMemo(() => {
+    if (!isDefined(value.topNBuckets)) {
+      return null;
+    }
+    return validateTopNBucket(value.topNBuckets);
```

Comment:
Validation of the form was already in place https://github.com/elastic/kibana/blob/51235a9c7a53d1a8ef9340f01c028389b9dda61e/x-pack/plugins/ml/public/alerting/anomaly_detection_rule/register_anomaly_detection_rule.tsx#L104

Added an error state to the form control itself here.